lib/spack/spack/build_systems/intel.py

```diff
@@ -61,132 +118,1108 @@ class IntelPackage(PackageBase):
+                d = unversioned_dirname
+
+        debug_print(d)
+        return Prefix(d)
+
+    def normalize_path(self, component_path, component_suite_dir=None,
+                       relative=False):
+        '''Returns the absolute or relative path to a component or file under a
+        component suite directory.
+
+        Intel's product names, scope, and directory layout changed over the
+        years.  This function provides a unified interface to their directory
+        names.
+
+        Parameters:
+
+            component_path (str): a component name like 'mkl', or 'mpi', or a
+                deeper relative path.
+
+            component_suite_dir (str): _Unversioned_ name of the expected
+                parent directory of component_path.  When absent or `None`, an
+                appropriate default will be used.  A present but empty string
+                `""` requests that `component_path` refer to `self.prefix`
+                directly.
+
+                Typical values: `compilers_and_libraries`, `composer_xe`,
+                `parallel_studio_xe`.
+
+                Also supported: `advisor`, `inspector`, `vtune`. The actual
+                directory name for these suites varies by release year. The
+                name will be corrected as needed for use in the return value.
+
+            relative (bool): When True, return path relative to self.prefix,
+                otherwise, return an absolute path (the default).
+        '''
+        # Design note: Choosing the default for `component_suite_dir` was a bit
+        # tricky since there better be a sensible means to specify direct
+        # parentage under self.prefix (even though you normally shouldn't need
+        # a function for that).  I chose "" to allow that case be represented,
+        # and 'None' or the absence of the kwarg to represent the most relevant
+        # case for the time of writing.
+        #
+        # In the 2015 releases (the earliest in Spack as of 2018), there were
+        # nominally two separate products that provided the compilers:
+        # "Composer" as lower tier, and "Parallel Studio" as upper tier. In
+        # Spack, we justifiably retcon both as "intel-parallel-studio@composer"
+        # and "...@cluster", respectively.  Both of these use the older
+        # "composer_xe" dir layout, as do their virtual package personas.
+        #
+        # All other "intel-foo" packages in Spack as of 2018-04 use the
+        # "compilers_and_libraries" layout, including the 2016 releases that
+        # are not natively versioned by year.
+
+        cs = component_suite_dir
+        if cs is None and component_path.startswith('ism'):
+            cs = 'parallel_studio_xe'
+
+        v = self.version_yearlike
+
+        # Glob to complete component_suite_dir. The alternatives could be
+        # merged, but using separate ones is clearer and likely more reliable.
+        suffix_glob = '_*.*.*'
```

Comment:
the problem is here: should not that be something like 
```
'_' + self.version
```
?
Using this in `self.normalize_suite_dir` you pick up all versions.

Reply:
with this fix I got right paths according to debug printing:
```
==> normalize_path.normalize_suite_dir:	/apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222
==> component_lib_dir.normalize_path:	/apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl
==> blas_libs.component_lib_dir:	/apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64
==> lapack_libs.blas_libs:	/apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64/libmkl_intel_lp64.so /apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64/libmkl_sequential.so /apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64/libmkl_core.so
==> lapack_libs.blas_libs:	/lib64/libpthread.so /lib64/libm.so /lib64/libdl.so
==> normalize_path.normalize_suite_dir:	/apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222
==> component_lib_dir.normalize_path:	/apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl
==> blas_libs.component_lib_dir:	/apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64
==> <lambda>.blas_libs:	/apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64/libmkl_intel_lp64.so /apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64/libmkl_sequential.so /apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64/libmkl_core.so
==> <lambda>.blas_libs:	/lib64/libpthread.so /lib64/libm.so /lib64/libdl.so
==> normalize_path.normalize_suite_dir:	/apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222
==> component_lib_dir.normalize_path:	/apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl
==> scalapack_libs.component_lib_dir:	/apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64
==> <lambda>.scalapack_libs:	/apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64/libmkl_scalapack_lp64.so /apps/intel/ComposerXE2018/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64/libmkl_blacs_intelmpi_lp64.so
```